serious bug

this is a serious bug, i can not sync data with mongo-connector, please see the log.

[mongo-connector.log.zip](https://github.com/mongodb-labs/mongo-connector/files/455161/mongo-connector.log.zip)_

InvalidBSON: 'utf8' codec can't decode byte 0xc0 in position 2: invalid start byte
2016-09-04 15:24:11,413 [ERROR] __main__:304 - MongoConnector: OplogThread <OplogThread(Thread-2, stopped 140569963648768)> unexpectedly stopped! Shutting down

we have the same question!

Posting the error from the log here so it's easier to view:
```python
2016-09-05 21:52:50,916 [ERROR] mongo_connector.util:92 - Fatal Exception
Traceback (most recent call last):
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\util.py", line 90, in wrapped
    func(*args, **kwargs)
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\oplog_manager.py", line 310, in run
    ns, timestamp)
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\util.py", line 32, in wrapped
    return f(*args, **kwargs)
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\doc_managers\elastic2_doc_manager.py", line 162, in update
    updated = self.apply_update(document['_source'], update_spec)
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\doc_managers\elastic2_doc_manager.py", line 112, in apply_update
    return super(DocManager, self).apply_update(doc, update_spec)
  File "c:\users\yaoshirong\appdata\local\programs\python\python35\lib\site-packages\mongo_connector\doc_managers\doc_manager_base.py", line 70, in apply_update
    wl = len(where)
TypeError: object of type 'NoneType' has no len()
2016-09-05 21:52:51,511 [ERROR] mongo_connector.connector:304 - MongoConnector: OplogThread <OplogThread(Thread-2, stopped 6792)> unexpectedly stopped! Shutting down
 ```

--- a/mongo_connector/doc_managers/doc_manager_base.py
+++ b/mongo_connector/doc_managers/doc_manager_base.py
@@ -67,8 +67,8 @@ class DocManagerBase(object):
                     if '.' in to_set:
                         path = to_set.split(".")
                         where = _retrieve_path(doc, path[:-1], create=True)
-                        wl = len(where)
                         index = _convert_or_raise(where, path[-1])
+                        wl = len(where)
                         if isinstance(where, list) and index >= wl:
                             where.extend([None] * (index + 1 - wl))
                         where[index] = value

